resolwe/flow/serializers/fields.py

```diff
@@ -0,0 +1,51 @@
+"""Resolwe custom serializer fields."""
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.encoding import smart_text
+
+from rest_framework.relations import RelatedField
+
+from resolwe.permissions.shortcuts import get_objects_for_user
+
+
+class ResolweSlugRelatedField(RelatedField):
+    """
+    Resolwe specific implementation of SlugRelatedField.
+
+    A read-write field that represents the target of the relationship
+    by a unique combination of 'slug' and 'version' attributes.
+
+    This is a modification of rest_framework.relations.SlugRelatedField.
+    Since slug is not unique, (but combination of slug and version is),
+    we filter objects by slug, by permissions and return object with
+    highest version.
+
+    """
+
+    default_error_messages = {
+        'does_not_exist': ('Invalid {model_name} {slug_name} "{value}" - object does not exist.'),
+        'invalid': ('Invalid value.'),
+    }
+
+    def __init__(self, slug_field=None, **kwargs):
+        """Initialize attributes."""
+        assert slug_field is not None, 'The `slug_field` argument is required.'
+        self.slug_field = slug_field
+        super(ResolweSlugRelatedField, self).__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        """Convert to internal value."""
+        user = getattr(self.context.get('request', None), 'user', None)
```

Comment:
You should probably swap Django's `AnonymousUser` for guardian's one, as it is done [here](https://github.com/genialis/resolwe/pull/504/commits/644739a977cc695a7b856e01775159e65aa7c490#diff-363866a33132f2ba0b58211776554c4dR27).
How does `get_objects_for_user` work with `None` as the user?

Reply:
Thanks for pointing this out: `get_objects_for_user` will fail if `user=None`. What exactly do you mean by "swap Django's AnonymousUser for guardian's one"? I don't understand. What is the difference?

Reply:
`AnonymouseUser` in Django is a different class than the normal `User`, so you cannot permissions to it. Guardian solves this with creating an actual user object, which is treated as a public user.

Actually tis may already be solved in `get_objects_for_user`, but please make sure that it works as intended.